polkadot/runtime/parachains/src/paras/mod.rs

```diff
@@ -1221,6 +1221,15 @@ const INVALID_TX_BAD_VALIDATOR_IDX: u8 = 1;
 const INVALID_TX_BAD_SUBJECT: u8 = 2;
 const INVALID_TX_DOUBLE_VOTE: u8 = 3;
 
+/// This is intermediate "fix" for this issue:
+/// https://github.com/orgs/paritytech/projects/119/views/20?pane=issue&itemId=66568838
+///
+/// It does not actually fix it, but makes the worst case better. Without that limit someone
+/// could completely DoS the relay chain by registering a ridiculously high amount of paras.
+/// With this limit the same attack could lead to some parachains ceasing to being able to
```

Comment:
Actually, Snowbridge just needs to verify parachain headers for BridgeHub (Para 1002). We don't care about parachains.

Reply:
Interesting. We could then limit to just system chains for now. For off-chain XCMP we need all though, but that is further down the road.